azkaban-common/src/main/java/azkaban/executor/container/watch/FlowStatusManagerListener.java

```diff
@@ -238,7 +240,10 @@ private void updatePodStatus(AzPodStatusMetadata event) {
       logger.info(format(
           "Flow execution-id %d for pod %s does not have a final status in database and will be "
               + "finalized.", executionId, event.getPodName()));
-      final String reason = "Flow Pod execution was completed.";
+      final String reason = "Flow pod execution was completed.";
+      if (finalStatus == Status.EXECUTION_STOPPED) {
```

Comment:
Looks like the argument `finalStatus` will be meaningful only when its value is `EXECUTION_STOPPED` and will be completely ignored otherwise. 
How about one of the following alternatives:
(1) Don't ignore the argument and if the flow is being finalized then always use the given `finalStatus` 
(2) Change the argument type to `Optional<Status> finalStatus` and use the argument only when `finalStatus.isPresent() == true`